Sources/Purchasing/Purchases/PurchasesType.swift

```diff
@@ -1008,6 +1009,35 @@ public protocol PurchasesSwiftType: AnyObject {
 
     #endif
 
+    /**
+     * Use this method only if you already have your own IAP implementation using StoreKit 2 and want to use
+     * RevenueCat's backend. If you are using StoreKit 1 for your implementation, you do not need this method.
+     *
+     * You only need to use this method with *new* purchases. Subscription updates are observed automatically.
+     *
+     * #### Example:
+     *
+     * ```swift
+     * let result = try await product.purchase(options: options)
+     * Purchases.shared.handleObserverModeTransaction(purchaseResult: result)
+     * ```
+     *
+     * - Note: You need to finish the transaction yourself after calling this method.
+     *
+     * - Parameter purchaseResult: The ``StoreKit.Product.PurchaseResult`` of the product that was just purchased.
+     *
+     * - Throws: An error of type ``ErrorCode`` is thrown if a failure occurs while handling the purchase.
+     *
+     * - Returns: A ``StoreTransaction`` if there was a transacton found and handled for the provided product ID.
+     *
+     * - Important: This should only be used if you have enabled observer mode during SDK configuration using 
+     * ``Configuration/Builder/with(observerMode:)``
+     */
+    @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+    func handleObserverModeTransaction(
+        purchaseResult: StoreKit.Product.PurchaseResult
```

Comment:
I think the swifty way would be
```suggestion
        _ purchaseResult: StoreKit.Product.PurchaseResult
```